book/HAS/hybrid-application-service-design.md

```diff
@@ -0,0 +1,88 @@
+# Hybrid Application Service (HAS) Design
+
+## Application CR Create
+
+HAC collects the information required for creating applications. An application is a collection of one or more components. Required information to be collected can be found in the API [doc](). When all the application information is available, HAC will create the Application CR to trigger the application creation. The HAS Application CR Operator will:
+1. Validate all the required fields in the Application CR
+2. If the app model repository and the GitOps repository repos are specified:
+   1. It will validate if the specified repo exists.
+3. If any of those are not specified:
+   1. it will generate a repo under the org https://github.com/redhat-appstudio-appdata
+   2. Update the application CR with the generated repo information
+   3. Triggers the creation of the GitOps repository for the generated repo
+   4. Create the application level devfile under the Application CR status
+
+## Component CR Create
+
+### Importing a Devfile Sample
+HAC will prompt the user to select a sample (retrieved via the Devfile Registry REST API) to choose from. The Git repository corresponding to this sample (such as https://github.com/devfile-samples/devfile-sample-code-with-quarkus) will be specified when creating the Component CR.
+
+### Importing from Git
+HAC will prompt the user to provide a Git repository.
+A ComponentDetectionQuery CR will be created, it will:
+1. If a devfile is present, then:
+   1. If the devfile is valid and has the outerloop Kubernetes component and the deploy command, the information in the Devfile will be returned
```

Comment:
If the devfile is valid but does not have the outerloop Kubernetes or the deploy command, does an error get returned?  i.e. devfile must support outerloop according to these specifications

Reply:
No error gets returned but the devfile will be ignored and we will match a devfile from the registry. The next point 2, specifies what happens if its not valid.